test/user.test.js

```diff
@@ -1919,6 +1920,89 @@ describe('User', function() {
             expect(token).to.exist();
           });
       });
+
+      it('forwards the "options" argument to user.save() ' +
+        'when adding verification token', function() {
+        const ctxOptions = {testFlag: true};
+        const verifyOptions = {
+          type: 'email',
+          from: 'noreply@myapp.org',
+        };
+
+        // before save operation hook to capture ctx.options when saving
+        // verification token in user instance
+        User.observe('before save', function(ctx, next) {
+          if (!ctx.isNewInstance) {
+            // not checking equality since other properties are added by user.save()
+            expect(ctx.options).to.contain({testFlag: true});
+          }
+          next();
+        });
+
+        return User.create({email: 'test@example.com', password: 'pass'})
```

Comment:
This block is repeated in every test, let's extract it into a shared piece please. At minimum, extract the credentials. I think it would be even better to move `User.create` into a `beforeEach` hook.

```js
let user;
beforeEach(givenUser);

// tests ...

function givenUser() {
  return User.create(...).then(u => user = u);
}
```